hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3331,6 +3340,16 @@ func (r *HostedClusterReconciler) reconcileNetworkPolicies(ctx context.Context,
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile virt launcher policy: %w", err)
 		}
+		kvInfraCluster, err := r.discoverKubevirtClusterClient(ctx, hcluster)
+		if err != nil {
+			return err
+		}
+		egressFirewall := egressfirewall.VirtLauncherEgressFirewall(kvInfraCluster.namespace)
+		if _, err := createOrUpdate(ctx, kvInfraCluster.Client, egressFirewall, func() error {
+			return reconcileVirtLauncherEgressFirewall(egressFirewall)
+		}); err != nil {
+			return fmt.Errorf("failed to reconcile firewall to deny metadata server egress: %w", err)
+		}
```

Comment:
Now that i'm wrapping my head around this further in the context of external infra, i see a couple of more gaps.

1. the egress firewall needs a unique name.  This is what the hc.Spec.Platform.Kubevirt.GenerateID can help with. See an example here, https://github.com/openshift/hypershift/blob/f80f08035967f26054994a38f8567c63ea617915/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L809 

2. The egress firewall needs to be cleaned up by us somehow. look at the `func (r *HostedClusterReconciler) delete(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error) {` function. cleanup likely belongs in there. Our use case is kind of similar to the `cleanupOIDCBucketData` because we're deleting resources outside of the mgmt cluster. 

3. (Not related to this PR, so maybe we fix it in a future follow up) The network policy for virt launcher pods isn't using the correct client. i just noticed that because you put this code next to it :D

Reply:
> Now that i'm wrapping my head around this further in the context of external infra, i see a couple of more gaps.
> 
> 1. the egress firewall needs a unique name.  This is what the hc.Spec.Platform.Kubevirt.GenerateID can help with. See an example here, https://github.com/openshift/hypershift/blob/f80f08035967f26054994a38f8567c63ea617915/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L809

By design there can be only one egressfirewall object per namespace and the name MUST be "default", if not ovn-kubernetes throw an error

https://docs.openshift.com/container-platform/4.8/networking/ovn_kubernetes_network_provider/configuring-egress-firewall-ovn.html#nw-egressnetworkpolicy-object_configuring-egress-firewall-ovn,

> 2. The egress firewall needs to be cleaned up by us somehow. look at the `func (r *HostedClusterReconciler) delete(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error) {` function. cleanup likely belongs in there. Our use case is kind of similar to the `cleanupOIDCBucketData` because we're deleting resources outside of the mgmt cluster.

Understood.

> 3. (Not related to this PR, so maybe we fix it in a future follow up) The network policy for virt launcher pods isn't using the correct client. i just noticed that because you put this code next to it :D

Yep was wondering the same, I will prepare a follow up.

Reply:
> By design there can be only one egressfirewall object per namespace and the name MUST be "default", if not ovn-kubernetes throw an error

NOoooooooooooooooo :( okay. that changes things a bit. 

We need this fix for now, but I think we're going to have to walk this change back out of hypershift once the kubevirt rhcos lands. My concern is that we can't support this rule for GA (due to collisions with existing egress rules in namespaces, and lack of cleanup), but it's something we have to do for now.